File: perfromance_test/python/linklist/freq_prime.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class Linkedlist:
    def __init__(self):
        self.head = None
        
    # add node in linklist
    def add(self,data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
        else:
            temp = self.head
            while temp.next:
                temp = temp.next
            temp.next = new_node
            
            
        # is_prime check function
        
    def is_prime(self,num):
        if(num<2):
            return False
        else:
            for i in range(2, int(num ** 0.5)+1):
                if(num%i == 0):
                    return False
        return True
    
    #count frequency function
    
    def freq_prime(self):
        count = 0
        temp = self.head 
        while temp.next:
            if self.is_prime(temp.data):
                count += 1
            temp = temp.next
        return count
    
    def print_linledlist(self):
        temp = self.head
        while temp.next:
            print(temp.data,end=' --> ')
            temp = temp.next
        print()
if __name__ == "__main__":
    linked_list = Linkedlist()
        
    linked_list.add(11)
    linked_list.add(13)
    linked_list.add(4)
    linked_list.add(17)
    linked_list.add(15)
        
    
    print("linked list")
    linked_list.print_linledlist()
    
    
    # Calculating the frequency of prime numbers
    prime_count = linked_list.freq_prime()
    print(f"Frequency of prime numbers in the linked list: {prime_count}")
    

File: perfromance_test/java/linklist/freq_prime.java
// find the frequency of prime numeber in given linklist 
class freq_prime{
    // Node class to represent each node in the linked list
    static class Node {
        int data;
        Node next;

        // Constructor for creating a new node
        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }

    Node head;  // Head node of the linked list

    // Function to add a new node at the end of the list
    public void add(int data) {
        Node newNode = new Node(data);

        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newNode;
        }
    }

    // Function to check if a number is prime
    public boolean isPrime(int num) {
        if (num <= 1) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        return true;
    }

    // Function to count the number of prime numbers in the linked list
    public int countPrimes() {
        int primeCount = 0;
        Node temp = head;

        while (temp != null) {
            if (isPrime(temp.data)) {
                primeCount++;
            }
            temp = temp.next;
        }

        return primeCount;
    }

    public static void main(String[] args) {
        freq_prime list = new freq_prime();

        // Adding some values to the linked list
        list.add(11);
        list.add(15);
        list.add(3);
        list.add(18);
        list.add(17);

        // Counting the prime numbers in the linked list
        int primeFrequency = list.countPrimes();

        System.out.println("Frequency of prime numbers in the linked list: " + primeFrequency);
    }
}
